sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -587,13 +587,24 @@ fromClause
     ;
 
 aggregationClause
-    : GROUP BY groupingExpressions+=expression (',' groupingExpressions+=expression)* (
+    : GROUP BY groupingExpressionWithGroupingAnalytics+=groupByClause
+      (',' groupingExpressionWithGroupingAnalytics+=groupByClause)*
+    | GROUP BY groupingExpressions+=expression (',' groupingExpressions+=expression)* (
       WITH kind=ROLLUP
     | WITH kind=CUBE
     | kind=GROUPING SETS '(' groupingSet (',' groupingSet)* ')')?
```

Comment:
Why do we still need this def?

Reply:
> Why do we still need this def?

GROUP BY A, B grouping sets(a,  (a, b )
GROUP BY A, B `, `grouping sets(a, (a, b))
Both support , so we still need this.

Reply:
Do you mean  this line ?
```
  | GROUP BY kind=GROUPING SETS '(' groupingSet (',' groupingSet)* ')'
```

Reply:
> Do you mean this line ?

Yea, we already support the case in L611?

Reply:
> GROUP BY A, B , grouping sets(a, (a, b))

Ah, you mean `GROUP BY A, B  grouping sets(a, (a, b))`? It seems the parser cannot accpet `GROUP BY A, B , grouping sets(a, (a, b))`.

Reply:
> > GROUP BY A, B , grouping sets(a, (a, b))
> 
> Ah, you mean `GROUP BY A, B grouping sets(a, (a, b))`? It seems the parser cannot accpet `GROUP BY A, B , grouping sets(a, (a, b))`.

I mean we need  
```
| kind=GROUPING SETS '(' groupingSet (',' groupingSet)* ')')?
```
to support `group by a, b grouping sets(a, (a, b))`
and since we add `GROUPING SET` in   groupingExpressionWithGroupingAnalytics
so we  support `group by a, b, grouping sets(a, (a, b))`.

Reply:
> since we add GROUPING SET in groupingExpressionWithGroupingAnalytics
so we support group by a, b, grouping sets(a, (a, b))

Could you add tests?

Reply:
After this work done the I will change code about https://github.com/apache/spark/pull/30144 base on this pr, then we can support  
` GROUP BY A, B , grouping sets(a, (a, b))`
` GROUP BY A, B , cube(a, (a, b))` etc..

Reply:
> > since we add GROUPING SET in groupingExpressionWithGroupingAnalytics
> > so we support group by a, b, grouping sets(a, (a, b))
> 
> Could you add tests?

Since for https://github.com/apache/spark/pull/30212#discussion_r531393983
we just add this test in PlanParserSuite in this pr, ok ?